tests.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require "minitest/autorun"
+
+require_relative "lib/logdna.rb"
+require_relative "lib/logdna/client.rb"
+require_relative "test_server.rb"
+
+class TestLogDNARuby < Minitest::Test
+  @@log_line = "log line"
+
+  def get_options(port)
+    {
+      hostname: "rubyTestHost",
+      ip: "75.10.4.81",
+      mac: "00:00:00:a1:2b:cc",
+      app: "rubyApplication",
+      level: "INFO",
+      env: "PRODUCTION",
+      endpoint: "http://localhost:#{port}",
+      flush_interval: 1,
+      flush_size: 5,
+      retry_timeout: 1
+    }
+  end
+
+  def warn_method(port)
+    options = get_options(port)
+    logdna_thread = Thread.start do
+      logger = Logdna::Ruby.new("pp", options)
+      logger.warn(@@log_line)
+    end
+
+    server_thread = Thread.start do
+      server = TestServer.new
+      recieved_data = server.start_server(port)
+
+      assert_equal(recieved_data[:ls][0][:line], @@log_line)
+      assert_equal(recieved_data[:ls][0][:app], options[:app])
+      assert_equal(recieved_data[:ls][0][:level], "WARN")
+      assert_equal(recieved_data[:ls][0][:env], options[:env])
+    end
+
+    logdna_thread.join
+    server_thread.join
+  end
```

Comment:
There's a ton of repeated code here. I'd pull it out into a common function:

```
  def test_log_level(level, port, expected_output)
    options = get_options(port)
    logdna_thread = Thread.start do
      logger = Logdna::Ruby.new("pp", options)
      logger.send(level, @@log_line)
    end

    server_thread = Thread.start do
      server = TestServer.new
      recieved_data = server.start_server(port)

      assert_equal(recieved_data[:ls][0], expected_output)
    end

    logdna_thread.join
    server_thread.join
  end
```

And just call that with the right parameters instead of having so much shared logic between `warn_method`, `info_method`, etc.

Reply:
Our logger does not work this way. the `log` method does not take the level as the first parameter. It has to be a message. And the level is defined either by calling the method (warn, debug) or by passing the options. By doing it like this I test so the level can be defined as a method on a logger instance.

Reply:
> Our logger does not work this way. the log method does not take the level as the first parameter. It has to be a message.

I think you're not understanding the code I pasted - this is a Ruby thing. `send` is a standard Ruby call that's used to call a method dynamically. The following arguments are what's passed to the call. So if `level` here is "info", `logger.send(level, @@log_line)` resolves to `logger.info(@@log_line)`. And this is how you should do it.

Reply:
gotcha, I did not know this. I thought you were suggesting to use  the 'log' function. I'll make this change. Thanks!